src/core/options.ts

```diff
@@ -65,16 +65,17 @@ export function resolveOptions(options: Options, root: string): ResolvedOptions
   resolved.types = resolved.types || []
 
   resolved.root = root
-  resolved.transformer = options.transformer || getVueVersion(root) || 'vue3'
+  resolved.version = resolved.version ?? getVueVersion(root)
+  resolved.transformer = options.transformer || `vue${Math.trunc(resolved.version) as 2 | 3}`
   resolved.directives = (typeof options.directives === 'boolean')
     ? options.directives
     : !resolved.resolvers.some(i => i.type === 'directive')
         ? false
-        : getVueVersion(root) === 'vue3'
+        : resolved.version >= 3
   return resolved
 }
 
-function getVueVersion(root: string) {
+function getVueVersion(root: string): number {
   const version = getPackageInfoSync('vue', { paths: [root] })?.version || '3'
-  return version.startsWith('2.') ? 'vue2' : 'vue3'
+  return +(version.split('.').slice(0, 2).join('.'))
```

Comment:
I prefer to have this as an literal of `vue3`, `vue2.7`, `vue2.6` to explicitly saying the version we'd support.

Reply:
I think we could support vue < 2.6 as well (like 2.5)

Reply:
Yes. My point is that the current implementation allows arbitrary numbers to be passed. Like I can have `2.71`, `1`, `4`, or `Infinity`. This will leads to confusing and undefined behavior. Using the enum for explicit about the version we are supporting.

Reply:
I see, maybe we could restrict it by TypeScript `2 | 2.7 | 3`. `2` for Vue <= 2.7.

Reply:
It's a bit weird to say `2` stands for `<=2.6` - as technically we should say `2` will default to `2.7` 😅

Reply:
Hmmm, I think it's okay for Vue users. Since Volar has the same behavior (`target` of `tsconfig`).